### PR TITLE
[docs] move recipes to individual pages

### PR DESCRIPTION
**Pull request by marcysutton**
## Description

This is a replacement PR for https://github.com/gatsbyjs/gatsby/pull/18922, which was stalled for a while and created a really difficult merge. This one incorporates the outstanding feedback and updates links throughout the docs to the new recipes page.

Thanks to @kevee for your initial work on it!

### Documentation

Updated the [docs-templates](https://www.gatsbyjs.org/contributing/docs-templates/) page in the contributing docs to reflect the new structure.

## Related Issues

https://github.com/gatsbyjs/gatsby/issues/16980

### CODE REVIEW

**Comment by laurieontech** [comment]
This is so cool! Love the changes.

Built it locally and all looks good to me.

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -31,2362 +31,85 @@ See [docs templates](/docs/docs-templates/) in the contributing docs for more he
...
-      />
-    ))}
-    // highlight-end
-  </div>
-)
-```
-
-**Note**: This method can make it difficult to match images with `alt` text for accessibility. This example uses images with `alt` text included in the filename, like `dog in a party hat.jpg`.
-
-d. an image of a fixed size using the `fixed` field instead of
```
Comment (laurieontech) [inline docs/docs/recipes.md]:
```suggestion
- [Using a starter](/docs/recipes/working-with-starters#using-a-starter)
```

Comment (laurieontech) [inline docs/docs/recipes.md]:
```suggestion
- [Creating a new site using a theme starter](/docs/recipes/working-with-themes#creating-a-new-site-using-a-theme-starter)
```

Comment (laurieontech) [inline docs/docs/recipes.md]:
```suggestion
- [Building a new theme](/docs/recipes/working-with-themes#building-a-new-theme)
```

File: www/src/data/sidebars/doc-links.yaml [hunk 2]
```diff
@@ -14,6 +14,27 @@
       link: /starters/
     - title: Awesome Gatsby Resources
       link: /docs/awesome-gatsby-resources/
+    - title: Recipes
```
Comment (gillkyle) [inline www/src/data/sidebars/doc-links.yaml]: It looks like there's two links in the sidebar to recipes (the other  on Lines 9-10). Is that intentional?

image

Comment (marcysutton) [reply www/src/data/sidebars/doc-links.yaml]: Nope, probably a merge issue! The first standalone link can go away. I'll make that change shortly. Good catch! 👍

**Comment by AishaBlake** [comment]
I made a very quick change. I'm not seeing anything that hasn't already been caught. My impression is that, if there are additional errors or issues, they were already in the recipes and we can deal with any that arise separately.

**Comment by marcysutton** [comment]
Thanks for the feedback, team! I've updated with the PR with your changes, and it should be good to go. (woohoo!)

**Comment by laurieontech** [comment]
Wahoo! This looks awesome @marcysutton. Thanks for pulling together all these details.

**Pull request merged**